src/pseudopeople/dataset.py

```diff
@@ -0,0 +1,154 @@
+from pathlib import Path
+from typing import Any, List, Optional, Tuple, Union
+
+import numpy as np
+import pandas as pd
+from tqdm import tqdm
+from vivarium import ConfigTree
+
+from pseudopeople.configuration import Keys
+from pseudopeople.constants.metadata import DATEFORMATS
+from pseudopeople.constants.noise_type_metadata import COPY_HOUSEHOLD_MEMBER_COLS
+from pseudopeople.entity_types import ColumnNoiseType, NoiseType, RowNoiseType
+from pseudopeople.loader import load_data
+from pseudopeople.schema_entities import COLUMNS, Dataset
+from pseudopeople.utilities import coerce_dtypes, get_randomness_stream
+
+
+class DatasetData:
+    def __init__(
+        self,
+        dataset: Dataset,
+        input_data: Union[Path, pd.DataFrame],
+        user_filters: List[Tuple],
+        seed: Any,
+    ):
+        self.dataset = dataset
+        self.data = load_data(input_data, user_filters)
+        self.randomness = get_randomness_stream(self.dataset.name, seed, self.data.index)
+        self.missingness = self.is_missing(self.data)
+
+    def __bool__(self):
+        return not self.data.empty
+
+    def is_empty(self, column_name: str) -> bool:
+        """Returns whether the column is empty."""
+        return self.missingness[column_name].all()
+
+    def get_non_empty_index(self, required_columns: Optional[List[str]] = None) -> pd.Index:
+        """Returns the non-empty data."""
+
+        if required_columns is None:
+            non_empty_data = self.data.loc[~self.missingness.all(axis=1)]
+        else:
+            missingness_mask = self.missingness[required_columns].any(axis=1)
+            non_empty_data = self.data.loc[~missingness_mask, required_columns]
+        return non_empty_data.index
+
+    def get_noised_data(
+        self, configuration: ConfigTree, noise_types: List[NoiseType]
+    ) -> pd.DataFrame:
+        """Returns the noised dataset data."""
+        self.format_data()
+        self.noise_dataset(configuration, noise_types)
+        self.drop_extra_columns()
+        return self.data
+
+    def format_data(self) -> None:
+        """Formats the data to match the expected format for noising."""
+        self._reformat_dates_for_noising()
+        self.data = coerce_dtypes(self.data, self.dataset)
+
+    def noise_dataset(self, configuration: ConfigTree, noise_types: List[NoiseType]) -> None:
```

Comment:
This is copy/paste but the special casing has been removed due to the missingness attribute of the class.